src/main/java/edu/kis/powp/jobs2d/drivers/adapter/FlippingLineDriverAdapter.java

```diff
@@ -0,0 +1,74 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class FlippingLineDriverAdapter implements Job2dDriver {
+    private final int UNDEFINED_POINT_VALUE = -10000;
+    public static final int FLIP_VERTICALLY = 0;
+    public static final int FLIP_HORIZONTALLY = 1;
+    public static final int FLIP_BOTH_SIDES = 2;
+
+    private final ILine line;
+    private int flippedX = 0, flippedY = 0;
+    private int startX = UNDEFINED_POINT_VALUE, startY = UNDEFINED_POINT_VALUE;
+    private final String name;
+    private final int flippingDirection;
+    private final DrawPanelController drawController;
+
+    public FlippingLineDriverAdapter(DrawPanelController drawController, ILine line, String name, int flippingDirection) {
```

Comment:
Instead of working on DrawPanelController, transforming should be able to be applied to all available drivers (e.g., also LoggerDriver). To do this you need to use Decorator patter approach.